packages/rum/src/domain/record/browser.types.ts

```diff
@@ -0,0 +1,12 @@
+// TODO: remove this once typescript has been update to 4.8+
+declare global {
```

Comment:
I ended using a type because I could extend `(Document | ShadowRoot)` with the type and avoid casting as`(doc as unknown as WithAdoptedStyleSheets)` and only do `doc as WithAdoptedStyleSheets`